src/main/webapp/app/atlas/agent/src/agent/ai_agent.py

```
@@ -0,0 +1,53 @@
+                messages=self.memory,
+                temperature=0.3
+            )
+
+            reply = response.choices[0].message.content
+            if not reply:
+                raise ValueError("Empty response from OpenAI API")
+
+        except Exception as e:
+            # Remove the user message if API call fails
+            self.memory.pop()
+            raise RuntimeError(f"Failed to get response from OpenAI: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)